mm2src/coins/lp_coins.rs

```diff
@@ -2509,14 +2522,16 @@ impl CoinsContext {
         let mut coins = self.coins.lock().await;
         let mut platform_coin_tokens = self.platform_coin_tokens.lock();
 
-        if coins.contains_key(platform.ticker()) {
+        let platform_ticker = platform.ticker().to_owned();
+        if platform.is_available() && coins.contains_key(&platform_ticker) {
             return MmError::err(PlatformIsAlreadyActivatedErr {
                 ticker: platform.ticker().into(),
             });
         }
 
-        let platform_ticker = platform.ticker().to_string();
-        coins.insert(platform_ticker.clone(), platform);
+        if platform.is_available() {
+            coins.insert(platform_ticker.clone(), platform);
+        }
```

Comment:
This is really confusing, we check if a platform coin `is_available` then we add it to coins context if it is. I understand why it's needed though since the coin is marked available in a previous step in activation but was not saved to coins context yet. I believe the real reason we have to do this hoops is we don't have a good structure for coins and tokens and how they related to each other inside the coins context in the first place, but this is a refactor outside of this PR scope.
Just a suggestion that I want to know your opinion on, what if availability is not saved inside each coin struct and instead there is a struct
```rust
struct MmCoinStruct {
    pub coin: MmCoinEnum,
    pub available: Option<bool>,
}
```
This is struct is what gets added to coins context here instead of `MmCoinEnum` https://github.com/KomodoPlatform/atomicDEX-API/blob/57be79eab446a62496e5a48694005abb6c8386a5/mm2src/coins/lp_coins.rs#L2452
As I see it, the availability attribute is related to the whole `MmCoinEnum` and should be external to it and not part of it's internals. This is a big change that will touch upon each part of the code changes here, so we need to make sure it will make the code better before doing it.

Reply:
This is better way of dealing with coin states, will refactor it.